GitCommands/UserRepositoryHistory/LocalRepositoryManager.cs

```diff
@@ -42,6 +42,8 @@ public interface ILocalRepositoryManager : IRepositoryManager
         /// <returns>An awaitable task.</returns>
         [ContractAnnotation("repositoryHistory:null=>halt")]
         Task SaveFavouriteHistoryAsync(IEnumerable<Repository> repositoryHistory);
+
+        Task RemoveInvalidRepositoriesAsync(Func<string, bool> predicate);
```

Comment:
Please provide xml-doc and `ContractAnnotation`

Reply:
Done.